docs/JSON-RPC-API/trace-module-overview.md

```diff
@@ -0,0 +1,53 @@
+# "trace" Module Overview
+
+The trace module is for getting a deeper insight into transaction processing. It includes two sets of calls; the transaction trace filtering API and the ad-hoc tracing API. You can find the documentation for the supported methods [here](/JSON-RPC-API/modules/trace/).
+
+!!! Note "Full sync"
+    In order to use the Transaction-Trace Filtering API, core-geth must be fully synced using `--syncmode=full --gcmode=archive`. Otherwise, you can set the number of blocks to `reexec` back for rebuilding the state, though taking longer for a trace call to finish.
+
+## JSON-RPC methods
+
+### Ad-hoc Tracing
+
+The ad-hoc tracing API allows you to perform a number of different diagnostics on calls or transactions, either historical ones from the chain or hypothetical ones not yet mined.
+
+- [x] trace_call
+- [x] trace_callMany
+- [ ] trace_rawTransaction
+- [ ] trace_replayBlockTransactions
+- [ ] trace_replayTransaction
+
+### Transaction-Trace Filtering
+
+These APIs allow you to get a full externality trace on any transaction executed throughout the blockchain.
+
+- [x] trace_block
+- [x] trace_transaction
+- [x] trace_filter (doesn't support address filtering yet)
+- [ ] trace_get
+
+### Available tracers
+
+- `callTracerParity` Transaction trace.
+- `vmTrace` Virtual Machine execution trace. Provides a full trace of the VM’s state throughout the execution of the transaction, including for any subcalls. *(Not implemented yet)*
+- `stateDiffTracer` State difference. Provides information detailing all altered portions of the Ethereum state made due to the execution of the transaction.
+
+!!! Example "Example trace_* API method config (last method argument)"
+
+    ```js
+    {
+        "tracer": "stateDiffTracer",
+        "timeout: "10s",
+        "reexec: "10000",               // number of block to reexec back for calculating state
+        "parityCompatibleOutput": true  // for Ad-hoc Tracing only in order the response is nested as in OpenEthereum output
+    }
+    ```
+
+## "stateDiff" tracer differences with OpenEthereum
+
+1. **SSTORE** in some edge cases persists data in state but are not being returned on stateDiff storage results on OpenEthereum output. *(Happens only on 2 transactions on Mordor testnet)*.
```

Comment:
Citations of canonical chain data should probably include state information, like _as of block 1,500,214_.

Reply:
Done at 2e486f6